code/go/0chain.net/smartcontract/dbs/event/provider.go

```diff
@@ -68,15 +57,52 @@ func (p *Provider) BeforeCreate(tx *gorm.DB) (err error) {
 	return
 }
 
+func (edb *EventDb) updateProviderTotalStakes(providers []Provider, tablename string) error {
+	var ids []string
+	var stakes []int64
+	for _, m := range providers {
+		ids = append(ids, m.ID)
+		i, err := m.TotalStake.Int64()
```

Comment:
We don't need to do the int64 converting if the updater support uint64, and it should be easy to support.